ggshield/cmd/sca/scan.py

```diff
@@ -57,14 +51,119 @@ def scan_diff_cmd(
 
 
 @scan_group.command(name="all")
+@click.argument(
+    "directory",
+    type=click.Path(exists=True, readable=True, path_type=Path, file_okay=False),
+    required=False,
+)
 @click.pass_context
-@display_sca_beta_warning
 def scan_all_cmd(
     ctx: click.Context,
     directory: Optional[Path],
     **kwargs: Any,
-) -> int:
+) -> Optional[SCAScanAllOutput]:
     """
     Scan a directory for SCA vulnerabilities.
     """
-    return 0
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx)
+
+    result = sca_scan_all(ctx, directory)
+
+    # TODO, should be handled by an output_handler in the future
+    return result
+
+
+def update_context(ctx: click.Context) -> None:
+    """
+    Basic implementation, should be populated with required parameters later.
+    """
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    # Empty for now
+    ctx.obj["exclusion_regexes"] = set()
+
+
+def sca_scan_all(ctx: click.Context, directory: Path) -> Optional[SCAScanAllOutput]:
```

Comment:
```suggestion
def sca_scan_all(ctx: click.Context, directory: Path) -> SCAScanAllOutput:
```

Reply:
There is still a `return None` line 146, in case of error in the second API call, where the client returns a `Detail` instance.